src/System.Management.Automation/engine/DataStoreAdapter.cs

```diff
@@ -856,16 +856,9 @@ public bool Equals(PSDriveInfo drive)
 
             if ((drive1Object == null))
             {
-                if (drive2Object == null)
-                {
-                    // Since both drives are null, they are equal
-                    return false;
-                }
-                else
-                {
-                    // Since drive1 is null it is less than drive2 which is not null
-                    return false;
-                }
+                // Since both drives are null, they are equal
```

Comment:
This doesn't look right.  I believe it should be.
return (drive2Object != null);

Reply:
Oh, thanks, I didn't notice that the refactoring can go even further